backend/typescript/services/implementations/emailService.ts

```diff
@@ -20,31 +20,156 @@ class EmailService implements IEmailService {
     }
   }
 
-  async sendConfirmationEmail(to: string): Promise<void> {
+  async sendConfirmationEmail(
+    to: string,
+    registrantName: string,
+    campName: string,
+    sessionDates: string,
+    campers: { name: string; age: number }[],
+    registrantPhoneNumber: string,
+    campFee: number,
+    totalCampFees: number,
+    dropOffAndPickupFees: number,
+    totalPayment: number,
+    link: string,
+  ): Promise<void> {
     await this.sendEmail(
       to,
-      "Focus on Nature: Confirmation Email",
-      "dummy body",
+      "Focus on Nature Camp Registration - Confirmation Email",
```

Comment:
STOP! look at the file in your code editor pls, the line breaks look so ugly here